src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -1130,3 +1130,10 @@ func defaultTLSConfig() *TLSConfig {
 		NodeSecretRef:     nil,
 	}
 }
+
+func (s Cluster) SASLAuthorizationEnabled() bool {
+	return s.Spec.KafkaEnableAuthorization ||
```

Comment:
Wonder if this new flag is needed. Setting authenticationMethod on the listener may be enough.

Reply:
I followed @BenPope implementation from https://github.com/redpanda-data/redpanda/pull/5292

In operator we don't have unlimited listener list, so your comment seem right. We have opinionated subset of Redpanda configuration, so I would like to defer this decision to @vladoschreiner.